src/JuliusSweetland.OptiKey.Core/Services/KeyboardOutputService.cs

```diff
@@ -490,12 +491,20 @@ public void ProcessSingleKeyText(string capturedText)
             }
         }
 
-        public void ProcessSingleKeyPress(string key, KeyPressKeyValue.KeyPressType type, int delayMs = 0)
+        public async Task ProcessSingleKeyPress(string inKey, KeyPressKeyValue.KeyPressType type)
         {
-            // TODO: This is a stub for future use
-            throw new NotImplementedException();            
+            Log.InfoFormat("Processing input value {0} with key press type {1}", inKey, type);
+
+            FunctionKeys funKey;
+            Char chaKey;
+            if (Enum.TryParse(inKey, out funKey))
+                PublishKeyPress(funKey, type);
+            else if (char.TryParse(inKey, out chaKey))
+                PublishKeyPress(chaKey, type);
+            else
+                Log.InfoFormat("No virtual key code found for '{0}'", inKey);
```

Comment:
I'm using this method so I can pass the KeyPressType. I stripped it down from my previous version and redirect most of the handling to PublishKeyPress.